frontend/lib/identification/rectangular_progress_indicator_painter.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:math';
+
+import 'package:flutter/widgets.dart';
+
+class RectangularProgressIndicatorPainter extends CustomPainter {
+  const RectangularProgressIndicatorPainter({
+    this.valueColor,
+    this.value,
+  });
+
+  final Color valueColor;
+  final double value;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final strokeWidth = 6.0;
+    final radius = 12.0;
+    var splashDuration = 0.01;
+    var delay = 0.01;
+    var fadeOutDuration = 0.03;
+    final strokeRadius = radius - strokeWidth / 2;
+    final verticalLineLength = size.height - strokeWidth - 2 * strokeRadius;
+    final horizontalLineLength = size.width - strokeWidth - 2 * strokeRadius;
+    final quarterArcLength = pi * strokeRadius / 2;
+    final paint = Paint();
+    paint.style = PaintingStyle.stroke;
+    paint.strokeWidth = strokeWidth;
+    paint.color = valueColor;
+    paint.strokeCap = StrokeCap.round;
+
+    final totalPathLength = 2 * horizontalLineLength +
+        2 * verticalLineLength +
+        4 * quarterArcLength;
+    Path createPath() {
+      var path = Path();
+      path.moveTo(size.width / 2, strokeWidth / 2);
+      var remainingPath = max(
+          totalPathLength * (value - delay / 2 - splashDuration) / (1 - delay),
+          0.0);
+      {
+        // First half top line
+        path.relativeLineTo(-min(remainingPath, horizontalLineLength / 2), 0);
+        remainingPath = max(remainingPath - horizontalLineLength / 2, 0);
+      }
+      {
+        // Top left arc
+        var angle = min(remainingPath / quarterArcLength, 1.0) * pi / 2;
+        var dest =
+            Offset(-strokeRadius * sin(angle), strokeRadius * (1 - cos(angle)));
+        path.relativeArcToPoint(dest,
+            radius: Radius.circular(strokeRadius), clockwise: false);
+        remainingPath = max(remainingPath - quarterArcLength, 0);
+      }
+      {
+        // Vertical line left
+        path.relativeLineTo(0, min(remainingPath, verticalLineLength));
+        remainingPath = max(remainingPath - verticalLineLength, 0);
+      }
+      {
+        // Bottom left arc
+        var angle = min(remainingPath / quarterArcLength, 1.0) * pi / 2;
+        var dest =
+            Offset(strokeRadius * (1 - cos(angle)), strokeRadius * sin(angle));
+        path.relativeArcToPoint(dest,
+            radius: Radius.circular(strokeRadius), clockwise: false);
+        remainingPath = max(remainingPath - quarterArcLength, 0);
+      }
+      {
+        // Horizontal line bottom
+        path.relativeLineTo(min(remainingPath, horizontalLineLength), 0);
+        remainingPath = max(remainingPath - horizontalLineLength, 0);
+      }
+      {
+        // Bottom right arc
+        var angle = min(remainingPath / quarterArcLength, 1.0) * pi / 2;
+        var dest =
+            Offset(strokeRadius * sin(angle), -strokeRadius * (1 - cos(angle)));
+        path.relativeArcToPoint(dest,
+            radius: Radius.circular(strokeRadius), clockwise: false);
```

Comment:
```suggestion
        path.relativeArcToPoint(dest,
            radius: Radius.circular(strokeRadius), clockwise: false);
        remainingPath = max(remainingPath - quarterArcLength, 0);
```
I think you forgot something here

Reply:
![image](https://user-images.githubusercontent.com/16086491/110505612-9aacda80-80fe-11eb-97c8-5d135ae3ba4e.png)

Causing this critical bug 😱🤯